src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -162,7 +162,96 @@ internal class DaprClientGrpc : DaprClient
             }
         }
 
+        /// <inheritdoc/>
+        public override Task<BulkPublishResponse> BulkPublishEventAsync<TValue>(
+            string pubsubName,
+            string topicName,
+            IReadOnlyList<TValue> events,
+            string contentType,
+            CancellationToken cancellationToken = default)
+        {
+            return BulkPublishEventAsync(pubsubName, topicName, events, contentType, null, cancellationToken);
+        }
+        
+        /// <inheritdoc/>
+        public override Task<BulkPublishResponse> BulkPublishEventAsync<TValue>(
+            string pubsubName,
+            string topicName,
+            IReadOnlyList<TValue> events,
+            string contentType,
+            Dictionary<string, string> metadata,
+            CancellationToken cancellationToken = default)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(pubsubName, nameof(pubsubName));
+            ArgumentVerifier.ThrowIfNullOrEmpty(topicName, nameof(topicName));
+            ArgumentVerifier.ThrowIfNull(events, nameof(events));
+            return MakeBulkPublishRequest(pubsubName, topicName, events, contentType, metadata, cancellationToken);
+        }
+        
+        private async Task<BulkPublishResponse> MakeBulkPublishRequest<TValue>(
+            string pubsubName,
+            string topicName,
+            IReadOnlyList<TValue> events,
+            string contentType,
+            Dictionary<string, string> metadata,
+            CancellationToken cancellationToken)
+        {
+            if (String.IsNullOrEmpty(contentType))
+            {
+                contentType = Constants.ContentTypeApplicationJson;
+            }
```

Comment:
Again, from the definition of `BulkPublishRequestEntry` I took the content type to be per message. Is this not the case?

Reply:
Per discussion offline, removed it.